classes/media/FFProbe.php

```diff
@@ -129,27 +124,58 @@ private function getFilePath() {
 	 * Invoke ffprobe on the command line.
 	 *
 	 * @access private
-	 * @return boolean	Success
+	 * @return array	Meta Data
 	 */
 	private function invokeFFProbe() {
 		global $wgFFprobeLocation;
 
 		if (!file_exists($wgFFprobeLocation)) {
-			$this->metadata = [];
-			return false;
+			return [];
 		}
 
 		$json = shell_exec(escapeshellcmd($wgFFprobeLocation . ' -v quiet -print_format json -show_format -show_streams ') . escapeshellarg($this->getFilePath()));
 
 		$metadata = @json_decode($json, true);
 
 		if (is_array($metadata)) {
-			$this->metadata = $metadata;
-		} else {
-			$this->metadata = [];
-			return false;
+			return $metadata;
 		}
-		return true;
+
+		return [];
+	}
+
+	public function loadMetaData( string $select = 'v:0' ): bool {
+		$cache = MediaWikiServices::getInstance()->getMainWANObjectCache();
+		$cacheKey = $cache->makeGlobalKey( 'EmbedVideo', 'ffprobe', $this->filename, $select );
+		$ttl = ( $this->file instanceof \File || is_string( $this->file ) )
```

Comment:
Based on the DocBlocks for the constructor and property, `$file` should always be an instance of `File`. We need to decide whether to update the DocBlocks or to fix the code.